android/src/com/mapswithme/maps/MwmActivity.java

```diff
@@ -1970,9 +1933,44 @@ public void onShareLocationOptionSelected()
     shareMyLocation();
   }
 
-  public void onLayerItemClicked(@NonNull Mode mode)
+  @Override
+  public void onLayerItemClick(@NonNull Mode mode)
   {
     closeFloatingPanels();
-    mMapButtonsController.toggleMapLayer(mode);
+    if (mMapButtonsController != null)
+      mMapButtonsController.toggleMapLayer(mode);
+  }
+
+  @Override
+  public ArrayList<MenuBottomSheetItem> getMenuBottomSheetItems(String id)
+  {
+    if (id.equals(MAIN_MENU_ID))
+    {
+      ArrayList<MenuBottomSheetItem> items = new ArrayList<>();
+      items.add(new MenuBottomSheetItem(R.string.placepage_add_place_button, R.drawable.ic_plus, this::onAddPlaceOptionSelected));
+      items.add(new MenuBottomSheetItem(
+          R.string.download_maps,
+          R.drawable.ic_download,
+          getDownloadMapsCounter(),
+          this::onDownloadMapsOptionSelected
+      ));
+      mDonatesUrl = Config.getDonateUrl();
+      if (!TextUtils.isEmpty(mDonatesUrl))
+        items.add(new MenuBottomSheetItem(R.string.donate, R.drawable.ic_donate, this::onDonateOptionSelected));
+      items.add(new MenuBottomSheetItem(R.string.settings, R.drawable.ic_settings, this::onSettingsOptionSelected));
+      items.add(new MenuBottomSheetItem(R.string.share_my_location, R.drawable.ic_share, this::onShareLocationOptionSelected));
+      return items;
+    }
+    else if (id.equals(PLACEPAGE_MORE_MENU_ID))
+      return mPlacePageController.getMenuBottomSheetItems();
+    return null;
+  }
+
+  @Override
+  public Fragment getMenuBottomSheetFragment(String id)
+  {
+    if (id.equals(LAYERS_MENU_ID))
+      return new ToggleMapLayerFragment();
+    return null;
```

Comment:
Is it normal expected behavior to return null? It would make sense to document it with a signature.